### DISCUSSION TITLE
autocomplete widget showMore() or equivalent?

### DISCUSSION THREAD

**Discussion by kedodrill** [algolia/instantsearch#5917] [Q&A] [answered]
For clarification: I'm talking about the autocomplete widget, not Autocomplete.js. 

Hi there, I have successfully made an autocomplete search UI by trying to emulate the autocomplete in the [showcase](https://www.algolia.com/doc/guides/building-search-ui/widgets/showcase/js/). I want to allow the user to load more results if they hit the bottom of the results. I'm not seeing a showMore or hitsPerPage or something like that on autocomplete. I'm not sure how I would go about loading more results. I looked into using Autocomplete.js but I don't think it's exactly what I want. Any ideas or options here?

**Comment by kedodrill** [comment, marked as answer]
Hmm okay I was able to sort of fix this, but it's a little weird. For anyone who wants to do something similar, you can use `configure`. I made a variable to store that and added it to my search widgets. I added a `hitsPerPage` variable, then I increment it by 10 if a user gets to the end of the results, then I remove the configure widget and add a new one using the new hitsPerPage value. A little weird, but it works.

**Comment by Haroenv** [comment]
For showMore, you likely want to change hits for infiniteHits, and can use its refine function to show more items. Although I'd recommend for using a full search page as destination after an autocomplete, as you have much more place, also for filters etc.

**Comment by kedodrill** [comment]
@Haroenv is there a way to do that with the autocomplete widget? https://www.algolia.com/doc/api-reference/widgets/autocomplete/js/ this is the guide I followed and it has some custom render code. Is there a way to forward the autocomplete results to another widget?

For some background, I'm making an app that prices various products. The autocomplete is a dropdown to search for products, if clicked on they will be inserted into a table. So a full search page is a little weird for that I feel.

**Reply by Haroenv** [reply]
In that case, you should use `searchBox` and `infiniteHits` together, or `hits` and `pagination`. autocomplete the widget is not meant for paginated hits or infinite hits.

**Reply by kedodrill** [reply]
Interesting. Is there a way to get `searchBox` acting like `autocomplete`? Why is autocomplete not meant for infinite hits?

**Reply by Haroenv** [reply]
Because it's not advised to have many interactive elements in a popup, as it can be hard to reach with keyboard for example. What you can do is put the entire InstantSearch in a popup though, and just use search box and infinite hits together, and that should work just fine

**Reply by kedodrill** [reply]
Are there any other reasons why it's not good to have many interactive elements in a popup? I am using `keydown` on the search input to handle the keyboard. If the user gets to the last result and tries to go down again, it'll add 10 to the hitsPerPage. It's quite fast even with a hundred or so results loaded. Then if the user closes the autocomplete popup I just set hitsPerPage back to default.

I know this may be a slightly abnormal combo, but I do think that showMore could be beneficial to autocomplete, perhaps with some default keybindings (or maybe just add that to the example and leave it up to the user).

It is true that I could put this in a popup with a search box and infinite hits. In my case I'm trying to replicate a very old application for my company; I'm trying to not change the UI very much.
